src/java.desktop/unix/native/libawt_xawt/awt/screencast_pipewire.c

```diff
@@ -0,0 +1,910 @@
+    LOAD_SYMBOL(fp_pw_stream_disconnect, "pw_stream_disconnect");
+    LOAD_SYMBOL(fp_pw_stream_destroy, "pw_stream_destroy");
+    LOAD_SYMBOL(fp_pw_init, "pw_init");
+    LOAD_SYMBOL(fp_pw_context_connect_fd, "pw_context_connect_fd");
+    LOAD_SYMBOL(fp_pw_core_disconnect, "pw_core_disconnect");
+    LOAD_SYMBOL(fp_pw_context_new, "pw_context_new");
+    LOAD_SYMBOL(fp_pw_thread_loop_new, "pw_thread_loop_new");
+    LOAD_SYMBOL(fp_pw_thread_loop_get_loop, "pw_thread_loop_get_loop");
+    LOAD_SYMBOL(fp_pw_thread_loop_signal, "pw_thread_loop_signal");
+    LOAD_SYMBOL(fp_pw_thread_loop_wait, "pw_thread_loop_wait");
+    LOAD_SYMBOL(fp_pw_thread_loop_accept, "pw_thread_loop_accept");
+    LOAD_SYMBOL(fp_pw_thread_loop_start, "pw_thread_loop_start");
+    LOAD_SYMBOL(fp_pw_thread_loop_stop, "pw_thread_loop_stop");
+    LOAD_SYMBOL(fp_pw_thread_loop_destroy, "pw_thread_loop_destroy");
+    LOAD_SYMBOL(fp_pw_thread_loop_lock, "pw_thread_loop_lock");
+    LOAD_SYMBOL(fp_pw_thread_loop_unlock, "pw_thread_loop_unlock");
+    LOAD_SYMBOL(fp_pw_properties_new, "pw_properties_new");
+
+    return TRUE;
+
+    fail:
+        dlclose(pipewire_libhandle);
+        pipewire_libhandle = NULL;
+    return FALSE;
+}
+
+void storeRestoreToken(const gchar* oldToken, const gchar* newToken) {
+
+    JNIEnv* env = (JNIEnv *) JNU_GetEnv(jvm, JNI_VERSION_1_2);
+    DEBUG_SCREENCAST("saving token, old: |%s| > new: |%s|\n", oldToken, newToken);
+    if (env) {
+        jstring jOldToken = NULL;
+        if (oldToken) {
+            jOldToken = (*env)->NewStringUTF(env, oldToken);
+            EXCEPTION_CHECK_DESCRIBE();
+            if (!jOldToken) {
+                return;
+            }
+        }
+        jstring jNewToken = (*env)->NewStringUTF(env, newToken);
+        EXCEPTION_CHECK_DESCRIBE();
+        if (!jNewToken) {
+            (*env)->DeleteLocalRef(env, jOldToken);
+            return;
+        }
+
+        jintArray allowedBounds = NULL;
+        if (screenSpace.screenCount > 0) {
+            allowedBounds = (*env)->NewIntArray(env, screenSpace.screenCount*4);
+            EXCEPTION_CHECK_DESCRIBE();
+            if (!allowedBounds) {
+                (*env)->ExceptionClear(env);
+                return;
+            }
+            jint* elements = (*env)->GetIntArrayElements(env, allowedBounds, NULL);
+            EXCEPTION_CHECK_DESCRIBE();
+            if (!elements) {
+                return;
+            }
+
+            gboolean failed = false;
+
+            for (int i = 0; i < screenSpace.screenCount; ++i) {
+                GdkRectangle bounds = screenSpace.screens[i].bounds;
+                elements[4 * i] = bounds.x;
+                elements[4 * i + 1] = bounds.y;
+                elements[4 * i + 2] = bounds.width;
+                elements[4 * i + 3] = bounds.height;
+
+                if ((*env)->ExceptionCheck(env)) {
+                    (*env)->ExceptionDescribe(env);
+                    (*env)->ExceptionClear(env);
```

Comment:
same here

Reply:
Actually, these checks are not needed here at all, this is an oversight, there was a Java function call before.
The check after ReleaseIntArrayElements is also unnecessary.